### PR TITLE
Additional entry point for review comments

### LINKED ISSUE #2077: additional entry point for review comments
We got some feedback from a user, via Twitter:

https://twitter.com/flaximus_prime/status/1118209612602400775

"In the list of open PRs there’s a button to jump right to the reviews of the checked-out PR; this is good.

 I want a similar way to jump right to the reviews for any open PR, without having to first open it in a tab. I find the workflows too different."

image

Seems like a reasonable suggestion to me -- not sure where to put the button though as the review list is already a bit crowded.

### PR DESCRIPTION
**Pull request by kuychaco**
**Please be sure to read the [contributor's guide to the GitHub package](https://github.com/atom/github/blob/master/CONTRIBUTING.md) before submitting any pull requests.**

### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* Suggestion: You can use checklists to keep track of progress for the sections on metrics, tests, documentation, and user research.

### Description of the Change

We received feedback asking to save users clicks when looking at reviews for non-checked out branches. See #2077.

This PR enables users to access reviews for _any_ PR listed in the GitHub tab (not just the checked out PR) via a "See reviews" context menu item.

I also took this opportunity to add a quick way to view the PR on dotcom via a "View on GitHub" context menu item.

### Screenshot/Gif

open-reviews

Fullscreen_4_18_19__2_14_PM

### Alternate Designs

Initially considered hiding the option behind a right-click context menu, but figured adding a kebab icon would make this more discoverable. The drawback is that the icon takes up some UI real-estate, but in my opinion it's small and discreet enough that it's fine. Open to feedback and other suggestions.

### Benefits

Currently, it is only possible to view review comments for the checked out PR from the PR list.

In order to view review comments for a PR that is **not** checked out the user has to click the desired PR in the PR list to open the PR detail pane item, open the "Files" tab, and click the "See reviews" button in the footer of the tab. 

With this change, we increase consistency by allowing the user to open reviews for _any_ PR directly from the PR list. Previously this was only possible for the checked out PR. 

### Possible Drawbacks

None really... the only mild concern is that the additional icon might increase visual noise.

### Applicable Issues

Fixes #2077 

### Metrics

Adding an event for 'open-reviews-tab'. 
Adding an event for 'open-issueish-in-browser' to collect information about the users leaving the editor for the browser.

### Tests

Tested `openOnGitHub` method along with `open-issueish-in-browser` event firing
Tested that `showActionsMenu` handler is called when the menu icon is clicked
Currently there is no good way to test Electron Menus.

### Documentation

N/A

### Release Notes

> Access review comments for any PR listed in the GitHub panel via an "ellipses" icon

### User Experience Research (Optional)

N/A

### CODE REVIEW

**Comment by aviflax** [comment]
👋 I’m the user who shared the feedback in #2077. @annthurium pointed me to this PR and I think the change looks great!!!

File: lib/controllers/issueish-list-controller.js [hunk 1]
```diff
@@ -51,12 +49,13 @@ export class BareIssueishListController extends React.Component {
...
     resultFilter: PropTypes.func,
     onOpenIssueish: PropTypes.func.isRequired,
+    onOpenReviews: PropTypes.func.isRequired,
     onOpenMore: PropTypes.func,
 
     emptyComponent: PropTypes.func,
     workspace: PropTypes.object,
     endpoint: EndpointPropType,
-    workingDirectory: PropTypes.string,
+    workdir: PropTypes.string.isRequired,
```
Comment (annthurium) [inline lib/controllers/issueish-list-controller.js]: I'm getting a PropTypes error in the console:

```
/Users/annthurium/github/github/node_modules/prop-types/checkPropTypes.js:20 Warning: Failed prop type: The prop `workdir` is marked as required in `BareIssueishListController`, but its value is `undefined`.
    in BareIssueishListController (created by ReactRelayQueryRenderer)
    in ReactRelayQueryRenderer (created by IssueishSearchContainer)
    in IssueishSearchContainer (created by IssueishSearchesController)
    in div (created by IssueishSearchesController)
    in IssueishSearchesController (created by RemoteController)
    in RemoteController (created by ReactRelayQueryRenderer)
    in ReactRelayQueryRenderer (created by ObserveModel)
    in ObserveModel (created by RemoteContainer)
    in RemoteContainer (created by GitHubTabView)
    in div (created by GitHubTabView)
    in div (created by GitHubTabView)
    in GitHubTabView (created by GitHubTabController)
    in GitHubTabController (created by ObserveModel)
    in ObserveModel (created by GitHubTabContainer)
    in GitHubTabContainer (created by GitHubTabItem)
    in GitHubTabItem (created by PaneItem)
    in PaneItem (created by RootController)
    in RootController
```

Is the workdir required by all places in the component tree the instantiate the `IssueishListController`?  If so we need to make sure they're all passing that prop.  Else mark it as not required.

Comment (kuychaco) [reply lib/controllers/issueish-list-controller.js]: Ooh good catch, forgot to clean this up when I lifted the logic to open `ReviewItem`s to the `IssueishSearchesController`... :fire:'d the unused prop

File: styles/issueish-list-view.less [hunk 2]
```diff
@@ -63,6 +63,11 @@
       text-align: center;
     }
 
+    &--menu {
+      color: @text-color-subtle;
+      transform: rotate(90deg);
```
[the remark this answers is not in the document]
Comment (kuychaco) [reply styles/issueish-list-view.less]: Sounds good! Thanks for making the change!

File: test/controllers/issueish-list-controller.test.js [hunk 3]
```diff
@@ -85,22 +85,40 @@ describe('IssueishListController', function() {
...
-    }));
+  describe('openOnGitHub', function() {
+    const url = 'https://github.com/atom/github/pull/2084';
+
+    it('calls shell.openExternal with specified url', async function() {
+      const wrapper = shallow(buildApp());
+      sinon.stub(shell, 'openExternal').callsArg(2);
+
+      await wrapper.instance().openOnGitHub(url);
+      assert.equal(shell.openExternal.args[0][0], url);
```
Comment (vanessayuenn) [inline test/controllers/issueish-list-controller.test.js]: Writing it like below would be a bit more readable:
```suggestion
      assert.isTrue(shell.openExternal.calledWith(url));
```

File: test/controllers/issueish-list-controller.test.js [hunk 4]
```diff
@@ -85,22 +85,40 @@ describe('IssueishListController', function() {
...
-    assert.isTrue(atomEnv.workspace.open.called);
-    assert.isTrue(reporterProxy.addEvent.calledWith('open-reviews-tab', {package: 'github', from: 'BareIssueishListController'}));
+    it('fires `open-issueish-in-browser` event upon success', async function() {
+      const wrapper = shallow(buildApp());
+      sinon.stub(shell, 'openExternal').callsArg(2);
+      sinon.stub(reporterProxy, 'addEvent');
 
-    atomEnv.destroy();
+      await wrapper.instance().openOnGitHub(url);
+      assert.equal(reporterProxy.addEvent.callCount, 1);
```
Comment (vanessayuenn) [inline test/controllers/issueish-list-controller.test.js]: `strictEqual` here would make the test more resilient than just `equal`.

File: test/controllers/issueish-list-controller.test.js [hunk 5]
```diff
@@ -85,22 +85,40 @@ describe('IssueishListController', function() {
...
+
+    it('handles error when openOnGitHub fails', async function() {
+      const wrapper = shallow(buildApp());
+      sinon.stub(shell, 'openExternal').callsArgWith(2, new Error('oh noes'));
+      sinon.stub(reporterProxy, 'addEvent');
+
+      try {
+        await wrapper.instance().openOnGitHub(url);
+      } catch (err) {
+        assert.equal(err.message, 'oh noes');
```
Comment (vanessayuenn) [inline test/controllers/issueish-list-controller.test.js]: `strictEqual` -- same as above

File: test/controllers/issueish-list-controller.test.js [hunk 6]
```diff
@@ -85,22 +85,40 @@ describe('IssueishListController', function() {
...
+      }
+      assert.equal(reporterProxy.addEvent.callCount, 0);
```
Comment (vanessayuenn) [inline test/controllers/issueish-list-controller.test.js]: and one more `strictEqual`

File: test/controllers/issueish-list-controller.test.js (hunk 4 above)
Comment (kuychaco) [reply test/controllers/issueish-list-controller.test.js]: Ah yes, overlooked that while copypastaing. Thanks!

**Comment by kuychaco** [comment]
That pending `atom.github [old]` check is irrelevant... it's an orphan left over from creating a new Azure pipeline to fix our Azure install. Gonna go ahead and merge this because we have all the green checks we need

**Pull request merged**